traits/tests/test_float_range.py

```diff
@@ -17,108 +17,102 @@
-
-    ice_temperature = Range(high=0.0)
-
-
-class ModelWithBaseRange(HasTraits):
-    """
-    Model containing simple BaseRange trait.
+def ModelFactory(name, RangeFactory):
     """
+    Helper function to create various similar model classes.
 
-    # Simple floating-point range trait.
-    percentage = BaseRange(0.0, 100.0)
+    Parameters
+    ----------
+    name : str
+        Name to give the created class.
+    RangeFactory : callable(*range_args, **range_kwargs) -> TraitType
+        Callable with the same signature as Range; this will be used
+        to create the model traits.
 
-    # Traits that exercise the various possiblities for inclusion
-    # or exclusion of the endpoints.
-    open_closed = BaseRange(0.0, 100.0, exclude_low=True)
+    Returns
+    -------
+    HasTraits subclass
+        Subclass containing various Range-like traits, for testing.
 
-    closed_open = BaseRange(0.0, 100.0, exclude_high=True)
-
-    open = BaseRange(0.0, 100.0, exclude_low=True, exclude_high=True)
+    """
 
-    closed = BaseRange(0.0, 100.0)
+    class ModelWithRanges(HasTraits):
+        """
+        Model containing various Range-like traits.
+        """
 
-    # Traits for one-sided intervals
-    steam_temperature = BaseRange(low=100.0)
+        # Simple floating-point range trait.
+        percentage = RangeFactory(0.0, 100.0)
 
-    ice_temperature = BaseRange(high=0.0)
+        # Traits that exercise the various possiblities for inclusion
+        # or exclusion of the endpoints.
+        open_closed = RangeFactory(0.0, 100.0, exclude_low=True)
 
+        closed_open = RangeFactory(0.0, 100.0, exclude_high=True)
 
-class ModelWithRangeCompound(HasTraits):
-    """
-    Model containing compound Range trait.
-    """
+        open = RangeFactory(0.0, 100.0, exclude_low=True, exclude_high=True)
 
-    # Range as part of a compound trait. This (currently)
-    # exercises a different code path in ctraits.c from the
-    # corresponding trait in ModelWithRange.
-    percentage = Either(None, Range(0.0, 100.0))
+        closed = RangeFactory(0.0, 100.0)
 
-    # Traits that exercise the various possiblities for inclusion
-    # or exclusion of the endpoints.
-    open_closed = Either(None, Range(0.0, 100.0, exclude_low=True))
+        # Traits for one-sided intervals
+        steam_temperature = RangeFactory(low=100.0)
 
-    closed_open = Either(None, Range(0.0, 100.0, exclude_high=True))
+        ice_temperature = Range(high=0.0)
 
-    open = Either(None, Range(0.0, 100.0, exclude_low=True, exclude_high=True))
+    ModelWithRanges.__name__ = name
 
-    closed = Either(None, Range(0.0, 100.0))
+    return ModelWithRanges
 
-    # Traits for one-sided intervals
-    steam_temperature = Either(None, Range(low=100.0))
 
-    ice_temperature = Either(None, Range(high=0.0))
+# A trait type that has a fast validator but doesn't accept any values.
+impossible = Instance(Impossible, allow_none=False)
```

Comment:
Just to confirm, this new trait is used to check that `None` is not an acceptable value. Right?

Reply:
Yes: I wanted a compound trait type that included a `Range` but didn't allow `None`.